File: inject_seo.py
import os
import glob

pages_dir = r"d:\Shop-Website\aufp\pages"
html_files = glob.glob(os.path.join(pages_dir, "*.html"))

inject_str = """
  <link rel="manifest" href="../manifest.json" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />
"""

for file in html_files:
    with open(file, 'r', encoding='utf-8') as f:
        content = f.read()
    
    if "manifest.json" not in content and "</head>" in content:
        content = content.replace("</head>", f"{inject_str}</head>")
        with open(file, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Updated {file}")

File: inject_seo.js
const fs = require('fs');
const path = require('path');

const pagesDir = path.join(__dirname, 'pages');
const files = fs.readdirSync(pagesDir).filter(f => f.endsWith('.html'));

const injectStr = `
  <link rel="manifest" href="../manifest.json" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />
`;

files.forEach(file => {
  const filePath = path.join(pagesDir, file);
  let content = fs.readFileSync(filePath, 'utf-8');
  if (!content.includes('manifest.json') && content.includes('</head>')) {
    content = content.replace('</head>', `${injectStr}</head>`);
    fs.writeFileSync(filePath, content, 'utf-8');
    console.log(`Updated ${file}`);
  }
});
